llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -1381,16 +1384,55 @@ void VPWidenCastRecipe::execute(VPTransformState &State) {
   }
 }
 
+void VPWidenCastEVLRecipe::execute(VPTransformState &State) {
+  unsigned Opcode = getOpcode();
+  auto Inst = cast<CastInst>(getUnderlyingInstr());
+  State.setDebugLocFrom(getDebugLoc());
+  assert(State.UF == 1 && "Expected only UF == 1 when vectorizing with "
+                          "explicit vector length.");
+
+  if (Inst->isCast()) {
```

Comment:
Is it possible to have anything else rather than cast here?

Reply:
done~